From c69d375c27f1044459bd52b364b55dc3c9a3f9c6 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Vianney=20le=20Cl=C3=A9ment=20de=20Saint-Marcq?=
 <vianney.leclement@essensium.com>
Date: Thu, 19 Mar 2015 13:54:08 +0100
Subject: [PATCH] Fix building with Linux 3.10 headers

The internal priv_netlink.h header redefines some macros and structs
that are now available from the kernel headers.  Remove them and include
the kernel header instead.
---
 src/drivers/priv_netlink.h | 29 ++---------------------------
 1 file changed, 2 insertions(+), 27 deletions(-)

diff --git a/src/drivers/priv_netlink.h b/src/drivers/priv_netlink.h
index 74d6ce5..9d5d552 100644
--- a/src/drivers/priv_netlink.h
+++ b/src/drivers/priv_netlink.h
@@ -9,6 +9,8 @@
 #ifndef PRIV_NETLINK_H
 #define PRIV_NETLINK_H
 
+#include <linux/netlink.h>
+
 /*
  * This should be replaced with user space header once one is available with C
  * library, etc..
@@ -45,18 +47,8 @@
 #define RTM_DELLINK (RTM_BASE + 1)
 #define RTM_SETLINK (RTM_BASE + 3)
 
-#define NLMSG_ALIGNTO 4
-#define NLMSG_ALIGN(len) (((len) + NLMSG_ALIGNTO - 1) & ~(NLMSG_ALIGNTO - 1))
 #define NLMSG_HDRLEN ((int) NLMSG_ALIGN(sizeof(struct nlmsghdr)))
-#define NLMSG_LENGTH(len) ((len) + NLMSG_ALIGN(sizeof(struct nlmsghdr)))
 #define NLMSG_SPACE(len) NLMSG_ALIGN(NLMSG_LENGTH(len))
-#define NLMSG_DATA(nlh) ((void*) (((char*) nlh) + NLMSG_LENGTH(0)))
-#define NLMSG_NEXT(nlh,len) ((len) -= NLMSG_ALIGN((nlh)->nlmsg_len), \
-			     (struct nlmsghdr *) \
-			     (((char *)(nlh)) + NLMSG_ALIGN((nlh)->nlmsg_len)))
-#define NLMSG_OK(nlh,len) ((len) >= (int) sizeof(struct nlmsghdr) && \
-			   (nlh)->nlmsg_len >= sizeof(struct nlmsghdr) && \
-			   (int) (nlh)->nlmsg_len <= (len))
 #define NLMSG_PAYLOAD(nlh,len) ((nlh)->nlmsg_len - NLMSG_SPACE((len)))
 
 #define RTA_ALIGNTO 4
@@ -71,23 +63,6 @@
 #define RTA_DATA(rta) ((void *) (((char *) (rta)) + RTA_LENGTH(0)))
 
 
-struct sockaddr_nl
-{
-	sa_family_t nl_family;
-	unsigned short nl_pad;
-	u32 nl_pid;
-	u32 nl_groups;
-};
-
-struct nlmsghdr
-{
-	u32 nlmsg_len;
-	u16 nlmsg_type;
-	u16 nlmsg_flags;
-	u32 nlmsg_seq;
-	u32 nlmsg_pid;
-};
-
 struct ifinfomsg
 {
 	unsigned char ifi_family;
-- 
2.3.4

